homeassistant/util/json.py

```diff
@@ -37,7 +35,7 @@ def load_json(filename: str, default: Union[List, Dict] = _UNDEFINED) \
     except OSError as error:
         _LOGGER.exception('JSON file reading failed: %s', filename)
         raise HomeAssistantError(error)
-    return {} if default is _UNDEFINED else default
+    return {} if default is None else default
```

Comment:
This not the same. By removing the [sentinel value](https://en.wikipedia.org/wiki/Sentinel_value), it is no longer possible to have `None` be the default value.

Reply:
There were no uses of `None` as default value except the place I changed in `storage.py`

Reply:
There was no use yet, as the storage helper is very new. I don't think that we should change this.

Reply:
The storage helper was the lone use. I mean't that no one else used `None` for `load_json`

By using sentinel value the function becomes not type-safe, as it accepts `object` and thus could return object if one were to pass it as default value.

Another option is to have `{}` as the default value and ignore that warning that generates with a comment that we are not changing it, nor do we return it as-is for the caller to modify.

Reply:
I don't feel too strongly I guess.

Any place that was using `util.json` or `util.yaml` needs to be rewritten to use the storage helper anyway.

Reply:
Actually, you could also make the sentinal value an empty dict, that way you can keep it all as dict. Comparison is still done with `is` so will work the same.